cmd/generate.go

```diff
@@ -81,6 +82,40 @@ func genProcessCluster(cmd *cobra.Command, clusterName string, p *ants.Pool) {
 		}
 	}
 
+	// list of components in cluster
+	for c, _ := range clusterComponents {
+		var clComps []string
+		clComps = append(clComps, c)
+
+		// only doing at cluster level as doing it in component process level was pain
+		if clusters == "" {
+			if components != "" {
+				break
+			}
+		}
+		d, err := os.Open(clusterDir)
+		if err != nil {
+			log.Fatal().Err(err).Msg("")
+		}
+		defer d.Close()
+		comDir, err := d.Readdirnames(-1)
```

Comment:
https://pkg.go.dev/os#File.Readdirnames
you better hope the directory isn't too large :P

using `-1` is a magic number; I would refactor to:

```
ALL_FILE_NAMES := -1
d.Readdirnames(ALL_FILE_NAMES)
```